File: Command/Java/src/command/turnOffLight.java
package command;

public class turnOffLight implements Command{
    private Light light;
    public turnOffLight(Light light){
        this.light = light;
    }


    @Override
    public void execute() {
        this.light.turnOff();
    }

    @Override
    public void undo() {
        this.light.turnOn();

    }
}

File: Command/Python/TurnOffLight.py
# -*- coding: utf-8 -*-
"""
Created on Wed May 18 21:03:45 2022

@author: Mahmoud Saeed
"""

import Command as command

class TurnOffLight(command.Command):
    light = None
    
    def __init__(self , light):
        self.light = light 
        
    def excute(self):
        self.light.turnoff()
    
    def undo(self):
        self.light.turnon()
        